bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncTableJob.java

```diff
@@ -0,0 +1,247 @@
+    if (!doCommandLine(otherArgs)) {
+      return 1;
+    }
+
+    Job job =
+        BigtableSyncTableAdapter.setupJobWithBigtable(
+            this,
+            otherArgs,
+            sourceBigtableProjectId,
+            sourceBigtableInstance,
+            sourceBigtableAppProfile,
+            targetBigtableProjectId,
+            targetBigtableInstance,
+            targetBigtableAppProfile);
+    if (!job.waitForCompletion(true)) {
+      LOG.info("Map-reduce job failed!");
+      return 1;
+    }
+    counters = job.getCounters();
+    return 0;
+  }
+
+  /**
+   * SyncTable command line parsing utility
+   *
+   * @param args
+   * @return
+   */
+  private boolean doCommandLine(final String[] args) {
+    String[] toolDefaultArgs = parseBigtableCommandLineArgs(args);
+    if (!BigtableSyncTableAdapter.parseCommandLine(this, NUM_ARGS, toolDefaultArgs)) return false;
+
+    // return false if options and args are not properly set
+    if (!BigtableSyncTableAdapter.verifyRequiredArgsSet(
+        this,
+        args,
+        sourceBigtableProjectId,
+        sourceBigtableInstance,
+        targetBigtableProjectId,
+        targetBigtableInstance)) return false;
+    return true;
+  }
+
+  /**
+   * parse bigtable configurations
+   *
+   * @param args
+   * @return
+   */
+  public String[] parseBigtableCommandLineArgs(String[] args) {
+    List<String> defaultArgs = new ArrayList<>();
+
+    for (int i = 0; i < args.length; i++) {
+      String cmd = args[i];
+
+      String sourceBigtableProjectIdKey = "--sourcebigtableproject=";
+      if (cmd.startsWith(sourceBigtableProjectIdKey)) {
+        sourceBigtableProjectId = cmd.substring(sourceBigtableProjectIdKey.length());
+        continue;
+      }
+
+      String sourceBigtableInstanceKey = "--sourcebigtableinstance=";
+      if (cmd.startsWith(sourceBigtableInstanceKey)) {
+        sourceBigtableInstance = cmd.substring(sourceBigtableInstanceKey.length());
+        continue;
+      }
+
+      String sourceBigtableAppProfileKey = "--sourcebigtableprofile=";
+      if (cmd.startsWith(sourceBigtableAppProfileKey)) {
+        sourceBigtableAppProfile = cmd.substring(sourceBigtableAppProfileKey.length());
+        continue;
+      }
+
+      final String targetBigtableProjectIdKey = "--targetbigtableproject=";
+      if (cmd.startsWith(targetBigtableProjectIdKey)) {
+        targetBigtableProjectId = cmd.substring(targetBigtableProjectIdKey.length());
+        continue;
+      }
+
+      final String targetBigtableInstanceKey = "--targetbigtableinstance=";
+      if (cmd.startsWith(targetBigtableInstanceKey)) {
+        targetBigtableInstance = cmd.substring(targetBigtableInstanceKey.length());
+        continue;
+      }
+
+      final String targetBigtableAppProfileKey = "--targetbigtableprofile=";
+      if (cmd.startsWith(targetBigtableAppProfileKey)) {
+        targetBigtableAppProfile = cmd.substring(targetBigtableAppProfileKey.length());
```

Comment:
There should be a better way to parse the command line in the package. have you checked other jobs?

Reply:
we're following the semantics from SyncTable. let me know if changes are required here.